web/src/app/users/UserContactMethodForm.tsx

```diff
@@ -123,6 +123,19 @@ export default function UserContactMethodForm(
             component={DestinationField}
             disabled={edit}
             destFieldErrors={errors.filter(isDestFieldError)}
+            mapOnChangeValue={(v: StringMap): FieldValueInput[] => {
+              return Object.entries(v).map(([k, v]) => ({
+                fieldID: k,
+                value: v,
+              }))
+            }}
+            mapValue={(v: FieldValueInput[]): StringMap => {
+              const map: StringMap = {}
+              v.forEach((fv) => {
+                map[fv.fieldID] = fv.value
+              })
+              return map
+            }}
```

Comment:
Potential cleanup:

```suggestion
            mapValue={(v: FieldValueInput[]): StringMap => (
              v.map((i) => ({
                [i.fieldID]: i.value
              }))
            )}
```

Reply:
This will return an array of objects like this `[{foo: 'bar'}, {bin: 'baz'}]` rather than `{foo: 'bar', bin: 'baz'}`

Reply:
Oops! I missed that in the variable definition.

Reply:
np, renamed the var for clairity